### ISSUE TITLE
Azcopy is failing to upload files from S3 compatible to azure storage account

### ISSUE THREAD

**Issue by Venkat201414** [Azure/azure-storage-azcopy#1852]
### Which version of the AzCopy was used? 
10.15
### Which platform are you using? (ex: Windows, Mac, Linux)
Windows

### What command did you run?
azcopy copy "https://s3api-core.xxx.com/xxx-container-dev/xxxx-4d97-bd6a-09c00f7b12b9" "https://xxxxxx.blob.core.windows.net/test?sv=sastoken"

### What problem was encountered?

failed to perform copy command due to error: cannot start job due to error: failed to scan path [xxxxxx] The filename, directory name, or volume label syntax is incorrect..

### How can we reproduce the problem in the simplest way?
Try to upload a file to S3Compatible. 

### Have you found a mitigation/solution?
Nope. We are blocked. Looking forward for a solution. We referred the some files in git (azcopy sourcecode), looks like azcopy does not recognize s3compatible link?

**Comment by IhorLeontiev** [comment]
Having the similar problem, if there any updates?

**Comment by Danielku15** [comment]
Seems azcopy v10 only supports AWS s3 storage, not S3 Compatible storage. Ignoring the whole test and documentation/help aspects the "only" changes needed are within https://github.com/Azure/azure-storage-azcopy/blob/main/common/s3URLParts.go where the URL parsing happens. 

I cloned the repo, adjusted the file to fit my URL parsing needs, made a build and bingo: copy works.

Not sure if it helps a lot but what I did was:

* change `s3EssentialHostPart` to the domain of my S3 storage
* change `s3HostPattern` to match my url schema. I have a pattern like `region.storage.provider.com` hence I modified the regex to match these parts
* change `NewS3URLParts` to fit the regex pattern matches. 
* `go build`
